### PR TITLE
Change the type of click, auxclick, and contextmenu to PointerEvent

### LINKED ISSUE #100: Should "click", "dblclick" and "contextmenu" events be PointerEvents?
@dtapuska just noticed that Edge sends `click` as an instance of `PointerEvent`, not just `MouseEvent`.  I think we've agreed before that this was a good idea, but I don't think we've talked about speccing it.

I think we can just add a short section to the PE spec "upgrading" `click` to `PointerEvent`.  Also `dblclick` and `contextmenu`.  @teddink, that's it right?  I didn't check drag events or anything.

### PR DESCRIPTION
**Pull request by NavidZ**
This is the followup change to WHATWG html spec and ui events spec pull requests to change the type of click, auxclick, and contextmenu events to pointerevents:

- https://github.com/w3c/uievents/pull/259
- https://github.com/whatwg/html/pull/5324

Closes https://github.com/w3c/pointerevents/issues/100

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/pointerevents/pull/317.html" title="Last updated on Dec 14, 2020, 12:43 AM UTC (ce9a14b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/pointerevents/317/90ac201...ce9a14b.html" title="Last updated on Dec 14, 2020, 12:43 AM UTC (ce9a14b)">Diff</a>

### CODE REVIEW

**Comment by patrickhlauke** [comment]
what's the status of this? just waiting for a review, or any more things happening upstream/in UIEvents?

**Comment by NavidZ** [comment]
The ui events part landed. The html spec part is sent. But Domenic suggested to wait until Chrome at least enables this and makes sure there is no regression. If so we will land the one in html spec. We can do either. Either merge now and if it turns out to cause some unrecoverable regressions then revert it. Or just wait until Chrome ships and confirms there is no bad regressions and then merge. I'll leave it up to you.

File: index.html
Comment (patrickhlauke) [inline index.html]: "but the dispatch process is going be what is defined in the original specification" > "but the dispatch process is going to match that of the original specification"

File: index.html
Comment (patrickhlauke) [inline index.html]: "stream such as voice recognition or keyboard the" > "stream - such as voice recognition software or a keyboard interaction - the"

**Review by patrickhlauke — CHANGES_REQUESTED** [review]

**Review by patrickhlauke — APPROVED** [review]

**Comment by patrickhlauke** [comment]
Just getting this ready but waiting for any tests/regression checks before merging.

**Comment by smaug----** [comment]
@NavidZ Is this now enabled by default in Chrome?

File: index.html
Comment (smaug----) [inline index.html]: This is not backwards compatible.

**Review by smaug---- — DISMISSED** [review]

**Comment by NavidZ** [comment]
Looking at the code it seems it is still under experimental flag.

@mustaqahmed @liviutinta were running some experiments to make sure it is backward compatible and it doesn't cause any issues. I haven't heard any bad news out of that so far. Could you provide any update you might have?

**Comment by liviutinta** [comment]
We are running an experiment, and gradually expanding its scope. There were no bugs logged yet. If there are no bugs logged by November we plan to start the shipping process. We believe the major risk will be around clientX/clientY being fractional. If there is any breakage, we plan to have a flag in the code to return clientX/clientY as long but keep click/auxclick/contextmenu events as PointerEvent.

File: index.html
Comment (mustaqahmed) [reply index.html]: I suggest ruling out only "zero" as a special case (and keep allowing negative ids implicitly):

> This identifier MUST be unique from all other active pointers in the top-level browsing context at the time, **except for the value zero which indicates the event is from a non-pointing device**.

Comment (patrickhlauke) [reply index.html]: took a slightly different approach to wording ... see if that makes sense though

**Review by patrickhlauke — APPROVED** [review]

File: index.html
Comment (mustaqahmed) [inline index.html]: I found "additional" a bit unclear.  What about this?

> For these events, all <code>PointerEvent</code> specific attributes (defined in this spec) other than <code>pointerId</code> and <code>pointertype</code> will have their default values.

File: index.html
Comment (mustaqahmed) [inline index.html]: Let's avoid the word "stream" which currently indicated a "batch" from a given pointerId (used twice).  The current spec uses two other terms to mean the concept we want here: "pointing device" or "pointer device".  So I suggest this:
> Otherwise, if they were generated by something other than a pointing device...

File: index.html
Comment (mustaqahmed) [inline index.html]: Please check my other suggestion to avoid the word "stream".

**Review by mustaqahmed — COMMENTED** [review]
I have a few more suggestions, sorry didn't spot them last time.

File: index.html
Comment (patrickhlauke) [reply index.html]: good stuff, yes. changed

File: index.html
Comment (patrickhlauke) [reply index.html]: done

File: index.html
Comment (patrickhlauke) [reply index.html]: done

**Comment by patrickhlauke** [comment]
@smaug---- @mustaqahmed @NavidZ @liviutinta one final review ... we happy with the latest wording/changes on this?

File: index.html
Comment (mustaqahmed) [inline index.html]: One more "stream" to get rid of.  Here is a suggestion:

> The <code>pointerId</code> and <code>pointertType</code> attributes of these events MUST be the same as the PointerEvents that caused these events if they are generated by a pointing device. 
 If the events are generated by a non-pointing device (such as voice recognition software or a keyboard interaction), <code>pointerId</code> MUST be 0 and <code>pointerType</code> MUST be an empty string.

**Review by mustaqahmed — APPROVED** [review]
Thanks Patrick for addressing my comments, LGTM.  I have one last suggestion; if you see a better way, please go ahead without waiting for my review again.

File: index.html
Comment (patrickhlauke) [reply index.html]: Done

File: index.html
Comment (smaug----) [inline index.html:316]: "value of zero is reserved to indicate events that were generated by something other than a pointing device." seems like backwards incompatible change.

Comment (NavidZ) [reply index.html:316]: You are right. But the reason we chose that was no other implementation uses 0 that at this point. So we believe it should be safe to do so now. Otherwise, technically all the numbers are valid in pointerid space. Actually now that we are at it for breaking compat we can force all valid pointerid ones to be positive numbers (again something that the current implementations already do AFAIK) so maybe we can reserver those negative numbers for other future features or something.
Having said that, unless we add yet another field to the prototype to indicate non-pointing device I cannot imagine a way to be fully backward compatible.

Comment (mustaqahmed) [reply index.html:316]: Good point...looks like we assumed `pointerId` must start with 1.  After your comment, I realized this behavior is mentioned only in the Note below.  I will file an issue now.

**Comment by smaug----** [comment]
There seem to be tests for this. <event>_is_a_pointerevent...

**Pull request merged**
